content/en/docs/how-tos/use-registries-in-build-farm.md

```diff
@@ -1,112 +1,239 @@
+```yaml
+# this is the Namespace in which your ServiceAccount will live
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    openshift.io/description: Automation ServiceAccounts for MyProject
+    openshift.io/display-name: MyProject CI
+  name: my-project
+---
+# this is the ServiceAccount whose credentials you will use
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: image-puller
+  namespace: my-project
+---
+# this grants your ServiceAccount rights to pull images
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: my-project-image-puller-binding
+  # the namespace from which you will pull images
+  namespace: ocp
+roleRef:
+  kind: ClusterRole
+  apiGroup: rbac.authorization.k8s.io
+  name: system:image-puller
+subjects:
+  - kind: ServiceAccount
+    namespace: my-project
+    name: image-puller
+---
+# the Group of people who should be able to manage this ServiceAccount
+kind: Group
+apiVersion: v1
+metadata:
+  name: my-project-admins
+users:
+  # these names are GitHub usernames
+  - bob
+  - tracy
+  - jim
+  - emily
+---
+# this allows the group of people admin access to the Namespace
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: my-project-admins-binding
+  namespace: my-project
+roleRef:
+  kind: ClusterRole
+  apiGroup: rbac.authorization.k8s.io
+  name: admin
+subjects:
+  - kind: Group
+    apiGroup: rbac.authorization.k8s.io
+    name: my-project-admins
+```
 
-We can then get the token of, e.g., `SA/builder` in `namespace/ci`:
+After the pull request is merged, you will be able to extract the pull token for your `ServiceAccount` using the `oc` CLI:
 
 ```bash
-$ namespace=ci
-$ sa=builder
-$ secret=$(oc get sa -n $namespace $sa -o json | jq -r '.imagePullSecrets[0].name')
-$ hostname=image-registry.openshift-image-registry.svc:5000
-$ oc get secret --namespace $namespace $secret -o json | jq '.data[".dockercfg"]' --raw-output | base64 --decode | jq --arg h $hostname 'with_entries(select(.key==$h))'
+$ oc get secrets --namespace my-project -o json | \
+    jq '.items[] | select(.type=="kubernetes.io/dockercfg") | select(.metadata.annotations["kubernetes.io/service-account.name"]=="image-puller") | .data[".dockercfg"]' --raw-output | \
+    base64 --decode | \
+    jq 'with_entries(select(.key == "registry.ci.openshift.org"))'
```

Comment:
We do not have this key `registry.ci.openshift.org` in the secret:

```
oc get secret builder-dockercfg-hph8f  -o json | jq -r '.data.".dockercfg"' | base64 -d | jq -r 'keys[]'
172.30.49.128:5000
image-registry.openshift-image-registry.svc.cluster.local:5000
image-registry.openshift-image-registry.svc:5000
```

The cause might be that we do not use the default route. Not sure.

Reply:
...